README.md

```diff
@@ -682,7 +682,7 @@ The assignment parameters used by the targeting feature variant assigner include
 }
 ```
 
-Based on the configured audiences for the variants included in this feature, if the application is executing under the context of a user named `Alec` then the value of the `Big` variant will be returned. If the application is executing under the context of a user named `Susan` then the value of the `Small` variant will be returned. If a user match does not occur, then group matches are evaluated. If the application is executing under the context of a user in the group `Ring0` then the `Big` variant will be returned. If the user's group is `Ring1` instead, then the user has  a 50% chance between being assigned to `Big` or `Small`. If there is not user match nor group match then the default rollout percentage is used. In this case, 80% of unmatched users will get the `Small` variant leaving the other 20% to get the `Big` variant since it is marked as the `Default`.
+Based on the configured audiences for the variants included in this feature, if the application is executed under the context of a user named `Alec` then the value of the `Big` variant will be returned. If the application is executing under the context of a user named `Susan` then the value of the `Small` variant will be returned, unless `Susan` is part of one of the groups or `DefaultRolloutPercentage` of `Big`. If a user match does not occur, then group matches are evaluated. If the application is executed under the context of a user in the group `Ring0` then the `Big` variant will be returned. If the user's group is `Ring1` instead, then the user has a 50% chance of being assigned to `Small`. If there is no user match nor group match, then the default rollout percentage is used. In this case, 80% of unmatched users will get the `Small` variant, leaving the other 20% to get the `Big` variant since it is marked as the `Default`.

```

Comment:
User override should take precedent over group override.

Reply:
Ah... I don't think that is how the Spring implementation works. And I thought I had the same implementation as you. I loop through the Variants, first checking User, then Group(s), then default percentage. If anything returns true, you get assigned that variant.

I would expect it to work this way as it is the same with Feature Flags, if anything returns true we stop processing and return true.

Reply:
Sounds like spring and .net have different behaviors. The point of specifying users directly is to ensure they get a given variant.

Reply:
Ok, so this was one of the things bugging me about the implementation, which is why I called it out. So, should it work like this?

Check the assigned users of each variant.
Check the assigned groups of each variant, we can't do by group as they don't have to be the same list.
Run defaultRolloutPercentage checks for each one
If still false, use the default

?

Reply:
Yes